tasties_app/views.py

```diff
@@ -1,22 +1,41 @@
 from django.shortcuts import render
 from tasties_app.models import Recipe, Rating
 from django.db.models import Avg
-from collections import OrderedDict
 
 
 def index(request):
     return render(request, 'tasties_app/index.html',)
 
 
 def base(request):
-    return render(request, 'tasties_app/base.html',)
+
+    search_input = request.POST.get('search')
+    if search_input:
+        recipe_object = Recipe.objects.filter(title=search_input)
+        if recipe_object:
+            return search_recipe(request, search_input)
+        else:
+            return recipes(request)
+
+    else:
+        return render(request, 'tasties_app/base.html')
 
 
 def recipes(request):
     recipes_list = Recipe.objects.all()
     recipes_with_ratings = {}
     for recipe in recipes_list:
         recipes_with_ratings[recipe] = Rating.objects.filter(recipe_id=recipe).aggregate(Avg('rating'))['rating__avg']
-    recipes_with_ratings = OrderedDict(sorted(recipes_with_ratings.items(), key=lambda x: x[1], reverse=True))
```

Comment:
This line is important, it sorts the recipes by the rating value.
Don't delete it.